components/script/dom/htmlimageelement.rs

```diff
@@ -1345,6 +1397,9 @@ pub enum ImageElementMicrotask {
         elem: DomRoot<HTMLImageElement>,
         generation: u32,
     },
+    DecodeTask {
+        elem: DomRoot<HTMLImageElement>,
```

Comment:
One last thing I thought about: there is currently a race condition between when the microtask runs, and the "in parallel" stuff described at 2.2.  For example if the request becomes broken, we will reject all promises, including the one for which the micro task hasn't run yet. 

I'm not sure this is actually a problem, but we may as well be consitent with the spec and:

1. In `Decode`, not store the promise on `pending_`, but rather store it here in the task. 
2. change `react_to_decode_image_sync_steps` as described above.

Reply:
I moved storing promise in task and tests are okey